pkg/resources/kafka/kafka.go

```diff
@@ -1203,20 +1232,38 @@ func getServiceFromExternalListener(client client.Client, cluster *v1beta1.Kafka
 // - prioritize upscale in order to allow upscaling the cluster even when there is a stuck RU
 // - prioritize missing broker pods to be able for escaping from offline partitions, not all replicas in sync which
 //		could stall RU flow
-func reorderBrokers(brokerPods corev1.PodList, desiredBrokers []v1beta1.Broker, brokersState map[string]v1beta1.BrokerState, controllerBrokerID int32) []v1beta1.Broker {
-	runningBrokers := make(map[string]struct{})
-	for _, b := range brokerPods.Items {
-		brokerID := b.GetLabels()["brokerId"]
-		runningBrokers[brokerID] = struct{}{}
-	}
-
+func reorderBrokers(runningBrokers, boundPersistentVolumeClaims map[string]struct{}, desiredBrokers []v1beta1.Broker, brokersState map[string]v1beta1.BrokerState, controllerBrokerID int32, log logr.Logger) []v1beta1.Broker {
 	brokersReconcilePriority := make(map[string]brokerReconcilePriority, len(desiredBrokers))
+	missingBrokerDownScaleRunning := make(map[string]struct{})
+	// logic for handling that case when a broker pod is removed before downscale operation completed
+	for id, brokerState := range brokersState {
+		_, running := runningBrokers[id]
+		_, pvcPresent := boundPersistentVolumeClaims[id]
+		ccState := brokerState.GracefulActionState.CruiseControlState
+		if !running && ccState == v1beta1.GracefulDownscaleRequired || ccState == v1beta1.GracefulDownscaleRunning {
```

Comment:
**Optional**: we **MAY** improve readability by adding parenthesis to emphasize the intended evaluation order which is now implicitly determined from operator precedence.

```suggestion
		if (!running && ccState == v1beta1.GracefulDownscaleRequired) || ccState == v1beta1.GracefulDownscaleRunning {
```